### ISSUE TITLE
Add / update a community channel or medium in contribution guidelines for discussion

### ISSUE THREAD

**Issue by nik72619c** [styled-components/styled-components#3656]
Currently, the [contribution guidelines](https://github.com/styled-components/styled-components/blob/e932038fffa1400c4ca68d04010b7d3f319298cb/CONTRIBUTING.md#how-can-we-help-you-get-comfortable-contributing) point contributors towards [spectrum](https://spectrum.chat/styled-components) for asking questions about contribution or asking mentors to guide them on the same. 

Since the spectrum chat is now read-only and might not suffice for the community to discuss `styled-components` questions for help or codebase, I would recommend we decide on a medium and make it official inside our styled-components documentation as well as our contribution guidelines. 

**Possible suggestions**
- [Discord](discord.gg/hfGUrbrxaU)
-[ Stack overflow](https://stackoverflow.com/questions/ask?tags=styled-components&newreg=70fdbe3a6a574ddebccd599eb75b12f4) (this is what we have right now but is not enforced via documentation, website, or contribution guidelines)
- [Github discussions](https://docs.github.com/en/discussions) (recommended since it requires less inter-application switch and keeps GitHub as the SSOT)
- [Slack](https://slack.com/intl/en-in/)

**Related links**
- [Posted the same question on stack overflow](https://stackoverflow.com/questions/70548644/is-spectrum-outdated-for-styled-components-contribution-and-discussion)
- [Github issue for adding discord as community channel](https://github.com/styled-components/styled-components-website/issues/764)

**Comment by nik72619c** [comment]
cc: @probablyup ☝🏼

**Comment by kitten** [comment]
Agreed; It's currently really hard to coordinate closing issues that have come up before or are questions that can be answered quickly. 

Stack Overflow is IMHO a last resort but currently what I point at in issue response templates, but I'd prefer a more topical destination. That said; there's simply not enough of a community yet to do so. I'm in favour of promoting and pointing to Discord more often to make it the primary destination and grow it even more. It could also be useful for coordinating new contributors in the hopes that we can get more traction on changes we've discussed but that the core team doesn't have time for.

I'm not too keen on opening GH Discussions for this repository, since it keeps questions out of issues but generally doesn't do so at a 100% success rate anyway, and isn't more likely to receive more credible replies apart from core team members 

So, I'd suggest to link to Discord consistently, use it in issue reply templates, add an issue template link to it which comes up before opening issues, and generally attempt at making it the first destination for contributors and users ✌️

**Comment by jasonwilliams** [comment]
@nik72619c Discord is and has been the official replacement for Spectrum so there’s no need to pick one. I already had a conversation with @probablyup about this almost a year ago.

I agree with @kitten that we should consistently point to Discord everywhere. Also we are in need of mods as there’s already 150 in there and I’m still the only admin so please ping me @kitten.

**Comment by probablyup** [comment]
I'm of the general opinion that turning on GitHub Discussions is a good thing, in addition to Discord. Not everyone uses Discord and does not want to install an additional piece of software that isn't search-indexable.

**Comment by nik72619c** [comment]
Yup, agreed!

**Comment by jasonwilliams** [comment]
I’m not against GD as I think we can reduce noise between us.
The 2 mediums complement each other as Discord is for more real-time discussion and better for noise. We have a lot of people asking “How do I do X” or “Why doesn’t this work” which doesn’t suit GitHub discussions very well. But for more larger questions it’s fine imo.

Next.js do this and it seems to work ok.

**Comment by kitten** [comment]
I do want to throw into this thread that I really don't see GH Discussions as an effective tool, except for pushing more issues somewhere else where they'll go unaddressed.
It's basically issues but somewhere else in terms of who'd reply to issues. Having gone through past issues there's a disproportionate amount that'll go unaddressed and would likely go unaddressed in Discussions too, so I'd love to highlight a single place where we're more likely to foster community engagement by new contributors ✌️

**Comment by nik72619c** [comment]
Right! I see that in terms of scalability both the platforms (discord and GitHub discussions) perform well. We would want to also look at which platform is more used and comfortable with the community. 
I agree with @probablyup that GH discussions help us prevent people from downloading additional software. On the other hand, I am personally more comfortable with discord rather than GH discussions! 😄

**Comment by nik72619c** [comment]
> Agreed; It's currently really hard to coordinate closing issues that have come up before or are questions that can be answered quickly.
> 
> Stack Overflow is IMHO a last resort but currently what I point at in issue response templates, but I'd prefer a more topical destination. That said; there's simply not enough of a community yet to do so. I'm in favour of promoting and pointing to Discord more often to make it the primary destination and grow it even more. It could also be useful for coordinating new contributors in the hopes that we can get more traction on changes we've discussed but that the core team doesn't have time for.
> 
> I'm not too keen on opening GH Discussions for this repository, since it keeps questions out of issues but generally doesn't do so at a 100% success rate anyway, and isn't more likely to receive more credible replies apart from core team members
> 
> So, I'd suggest to link to Discord consistently, use it in issue reply templates, add an issue template link to it which comes up before opening issues, and generally attempt at making it the first destination for contributors and users ✌️ (i.e. we should add contact links to issue creation: [FormidableLabs/urql@`771607e`/.github/ISSUE_TEMPLATE/config.yml#L2](https://github.com/FormidableLabs/urql/blob/771607e666fb9a40fb8fc8cc2c16cdd68f17cf78/.github/ISSUE_TEMPLATE/config.yml#L2))
> 
> cc @jasonwilliams

Yeah I agree to it! I too personallt feel discord is more sort of `informal` and friendly for frequent questioning and discussions

**Comment by kitten** [comment]
Another factor here to throw in is what's safe to give a go here. I feel that GitHub Discussions is hard to turn off. We're basically just throwing an attempt out there when we promote Discord more, i.e. we're investing in that community's growth.
That's something that, push comes to shove, can be moved. I don't see GitHub Discussions as the same. Once we enable it and we see low engagement and low community vibes, we cannot simply turn it back off, partly because of the indexability. I'd also argue that I don't see a lot of things as needing to be indexable. A lot of issues will and are duplicated and I have trouble myself finding old ones sometimes. In styled-components' current phase I mainly think I'd prioritise *community-building* over *indexed threads*, so to speak.

tl;Dr We have a large userbase, but not a large community. For several reasons, GitHub Discussions plays into our weaknesses and not strengths while other platforms would play into strengthening this situation instead.
Basically, if one day a "new Spectrum" comes around that's more like Discord, we can think of moving the community I'd want to build for our users sake.

**Comment by nik72619c** [comment]
@probablyup Let us know your thoughts here 😊 !

**Comment by agriffis** [comment]
Discord doesn't need to be downloaded. I run it in my browser—didn't even realize there was a standalone app!

I'm probably -1 on GH Discussions. The problem in a project like this one is that it functions as a low-rigor please-help-me catch-all. Contrast issues where we can push back and ask for a minimal reproducer on codesandbox or a repo. Also contrast Discord where such requests will pass by instead of accumulating.

It's not that I don't want to help people, but it's easy for people to post, "It's not working for me" and they typically need to do some work before I can help. Discussions provide that uneasy middle ground where it's hard to push back and hard to let go.

**Comment by nik72619c** [comment]
@probablyup Any decision on this yet?

**Comment by fatso83** [comment]
I can update #3682 to point to _something_ if you agree on some consensus.

**Comment by kitten** [comment]
@fatso83 I'm planning to update much more than that and look into issue forms and the issue picker to get more traffic through to the invite link. So, until then, we won't have to do much. I'll have to come back to @jasonwilliams though, so we can get everything all set up ❤️ It looks like Discord for instance has a community mode that'd work well for us. So, I'm pretty set on taking the initiave here and getting it all set up. Currently, I'm assuming there are no objections

**Comment by fatso83** [comment]
OK, that's great of course, but maybe just merge the PR to delete the current references to Spectrum then while you work on getting it done? _Working is better than perfect_ 😄 Better not to know where to go than getting sent to the wrong place.

**Comment by jasonwilliams** [comment]
We have quite a few people on the Discord server now and no one is complaining. It’s true it won’t be for everyone but it seems to be the closest we have to a consensus.

@kitten i would put updating any references as a higher priority than what we do on the server because that can always come later. If right now we have links going to spectrum that should be resolved.
The invite link may need to be updated if we want people to come in via a “welcome” channel, I’m not sure.

**Comment by agriffis** [comment]
I like the Discord, but I'm wondering if the welcome sequence is contributing. For me, it pushes interesting conversation off the screen, but I absolutely don't want to complain if somebody likes it. I just wanted to check :sweat_smile: 

![image](https://user-images.githubusercontent.com/50637/152578239-2f713e81-eca3-46da-bd7d-3cf6f4622b58.png)

**Comment by jasonwilliams** [comment]
Hey @agriffis I have no idea how to change that sadly. If you have some familiarity with this im happy for help. Please PM me.
It looks like, and i could be wrong here, changing the channel the welcomes happen means sending out a new invite link.

**Comment by quantizor** [comment]
Closing — this issue is stale (2+ years without resolution) and relates to an older version or has been superseded by v6.x changes. If this is still reproducible on the latest v6.x release, please open a new issue with a minimal reproduction.
